src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -0,0 +1,133 @@
+            string metricOperation = null,
+            string learningRateOperation = null,
+            float learningRate = 0.01f,
+            bool addBatchDimensionInput = false,
+            DnnFramework dnnFramework = DnnFramework.Tensorflow)
+        {
+            var options = new Options()
+            {
+                ModelLocation = modelPath,
+                InputColumns = inputColumnNames,
+                OutputColumns = outputColumnNames,
+                LabelColumn = labelColumnName,
+                TensorFlowLabel = tensorFlowLabel,
+                OptimizationOperation = optimizationOperation,
+                LossOperation = lossOperation,
+                MetricOperation = metricOperation,
+                Epoch = epoch,
+                LearningRateOperation = learningRateOperation,
+                LearningRate = learningRate,
+                BatchSize = batchSize,
+                AddBatchDimensionInputs = addBatchDimensionInput,
+                ReTrain = true
+            };
+
+            var env = CatalogUtils.GetEnvironment(catalog);
+            return new DnnEstimator(env, options, DnnUtils.LoadDnnModel(env, modelPath, true));
+        }
+
+        /// <summary>
+        /// Performs image classification using transfer learning.
+        /// </summary>
+        /// <param name="catalog"></param>
+        /// <param name="featuresColumnName"></param>
+        /// <param name="labelColumnName"></param>
+        /// <param name="outputGraphPath"></param>
+        /// <param name="scoreColumnName"></param>
+        /// <param name="predictedLabelColumnName"></param>
+        /// <param name="checkpointName"></param>
+        /// <param name="arch"></param>
+        /// <param name="dnnFramework"></param>
+        /// <param name="epoch"></param>
+        /// <param name="batchSize"></param>
+        /// <param name="learningRate"></param>
+        /// <param name="addBatchDimensionInput"></param>
+        /// <remarks>
+        /// The support for image classification is under preview.
+        /// </remarks>
+        public static DnnEstimator ImageClassification(
+            this ModelOperationsCatalog catalog,
+            string featuresColumnName,
+            string labelColumnName,
+            string outputGraphPath = null,
+            string scoreColumnName = "Scores",
+            string predictedLabelColumnName = "PredictedLabel",
+            string checkpointName = "_retrain_checkpoint",
+            Architecture arch = Architecture.ResnetV2101,
+            DnnFramework dnnFramework = DnnFramework.Tensorflow,
+            int epoch = 10,
+            int batchSize = 20,
+            float learningRate = 0.01f,
+            bool addBatchDimensionInput = false)
+        {
+            var options = new Options()
+            {
+                ModelLocation = arch == Architecture.ResnetV2101 ? @"DnnImageModels\Resnet101V2Tensorflow\resnet_v2_101_299.meta" : "",
```

Comment:
The TensorFlow model location and rest of the base TF model's parameters are hard-coded in the ImageClassification method.
This should be configurable depending on the chosen base-model/`Architecture`.

Even if we currently have a single base-model/`Architecture` (ResnetV2101) it could be coded as if we had more additional models such as Inceptionv3, which we should have soon since most of our samples use Inceptionv3, so we can compare performance and quality.

Reply:
They are not hardcoded, you can set them from API. @"DnnImageModels\Resnet101V2Tensorflow\resnet_v2_101_299.meta" will be downloaded to the bin location when you add the nuget for resnet.

Reply:
You're right, I only meant the Model location. But we might want to hide.
The only thing is that if the users chooses now  `Architecture.InceptionV3`, then the model location is blank, and it might get an exception.. 
But if we're adding InceptionV3 soon, I'm okay. I'll resolve this topic. 👍